### ISSUE TITLE
Extract plain text form html

### ISSUE THREAD

**Issue by humphd** [Seneca-CDOT/telescope#75]
We are building a bunch of pieces of our pipeline that will give us HTML, that is, a blog post from a user's feed.  What we also need for later parts of the analysis pipeline is the ability to work on plain text vs. html.

Let's add a parsing function that works like this:

- given some HTML as an argument
- parse the HTML using https://github.com/jsdom/jsdom into a DOM
- `querySelector()` for the element we want (maybe all of `document.body`?)
- once we have that `elem` we can obtain it's text using `elem.innerText` and return that from the function

We also need tests for this, which can happen in another issue/PR.  @PavanKKamra has expressed some interest in this, and he might be glad to have some others work with him on it.

**Comment by PavanKKamra** [comment]
Hello David, 

I would like to attempt this issue. 

Thank you, 
Pavan K.

**Comment by humphd** [comment]
OK @PavanKKamra, go for it.  Please ask questions in here, and keep us updated on your progress.

To get started, I would do the following:

- being a new branch `issue-75`
- add JSDOM using `npm install --save jsdom`
- add a new file to `src` named `text-parser.js` or something
- in your new file, create a function with this basic shape:

```js
const jsdom = require("jsdom");

const { JSDOM } = jsdom;

exports.run = async function (html) {
  // put your code here.
}
```

**Comment by Reza-Rajabi** [comment]
@humphd Do we need to close #16 now?

**Comment by Reza-Rajabi** [comment]
@humphd It looks similar to #16 to me.

**Comment by humphd** [comment]
@Reza-Rajabi it's a portion of what happens after #16 I think.  We can make #16 be specifically about just parsing from url to xml.

**Comment by humphd** [comment]
In #83, an initial implementation of this is being landed, so the tests can pass.  We can probably improve this a bit by changing the JSDOM parsing to use [`fragment`](https://github.com/jsdom/jsdom#fragment) parsing, and simplify the code a bit.

**Comment by humphd** [comment]
If we can't get this to happen with JSDOM, we may need to switch to a full browser via Puppeteer, see https://github.com/GoogleChrome/puppeteer/issues/3260#issuecomment-422081505.

**Comment by humphd** [comment]
NOTE for later: we can also use [`page.setContent(html)`](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagesetcontenthtml-options) to render a snippet of HTML in a real DOM, before calling into it to get `innerText`.
